### PR TITLE
[PHPCS] Provide Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Check the following items.

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

If you are adding a new runner:

- [] ~~Read and follow the [document](https://github.com/sider/runners/blob/master/docs/how-to-write-a-new-runner.md) for a new runner.~~
- [] ~~Add a new tool to [`analyzers.yml`](https://github.com/sider/runners/blob/master/analyzers.yml).~~
- [] ~~Run `bundle exec rake readme:generate` and commit the updated `README.md`.~~
- [] ~~Add a new tool to the [Dependabot configuration](https://github.com/sider/runners/blob/master/.github/dependabot.yml) if needed.~~
- [] ~~Provide option(s) that users can customize.~~
- [x] Write smoke test cases. E.g. all options, successful end, warnings, errors, etc.
- [] ~~Add a [CI setting](https://github.com/sider/runners/blob/master/.github/workflows/build.yml).~~
- [] ~~Write a [document](https://github.com/sider/sider-docs) and link the pull request -> sider/sider-docs#557~~

### CODE REVIEW

**Comment by sleeeto** [comment]
@ybiquitous Ready

File: images/code_sniffer/sider_recommended_code_sniffer.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="SiderRecPSR12">
```
Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer.xml:2]: [IMO] The abbreviation `Rec` has some meanings like `receipt`, `recipe`, `record`, and so on. So I think it's confusing. How about just `SiderPSR12`?

```suggestion
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="SiderPSR12">
```

Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer.xml:2]: [question] Does it make sense to add the XML schema like [the PSR1 ruleset](https://github.com/squizlabs/PHP_CodeSniffer/blob/HEAD/src/Standards/PSR1/ruleset.xml)?

```suggestion
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/squizlabs/PHP_CodeSniffer/HEAD/phpcs.xsd" name="SiderRecPSR12">
```

File: images/code_sniffer/sider_recommended_code_sniffer_ext.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    <rule ref="Generic.VersionControl.GitMergeConflict"/>
+    <rule ref="PEAR.Classes.ClassDeclaration"/>
+    <rule ref="PEAR.ControlStructures.ControlSignature"/>
+    <rule ref="PSR2.Namespaces.NamespaceDeclaration"/>
+    <rule ref="PSR2.Namespaces.UseDeclaration"/>
+    <rule ref="Squiz.Classes.LowercaseClassKeywords"/>
+    <rule ref="Squiz.Functions.FunctionDuplicateArgument"/>
+    <rule ref="Squiz.PHP.InnerFunctions"/>
+    <rule ref="Squiz.Scope.MemberVarScope"/>
+    <rule ref="Squiz.Scope.StaticThisUsage"/>
```
Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer_ext.xml:78]: [IMO] I feel readable if rules are sorted alphabetically.

File: images/code_sniffer/sider_recommended_code_sniffer.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <description>The Sider's recommended ruleset (SiderRecPSR12) provides a set of useful rules distilled from PSR12.</description>
```
Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer.xml:3]: [IMO] Including a ruleset name in the description seems needless to me.

```suggestion
    <description>The Sider's recommended ruleset provides a set of useful rules distilled from PSR12.</description>
```

File: images/code_sniffer/sider_recommended_code_sniffer_ext.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="SiderRecExt">
+    <description>The Sider's recommended ruleset extension (SiderRecExt) is a set of useful rules except PSR12, and is intended to be used in combination with SiderRecPSR12.</description>
```
Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer_ext.xml:3]: Also here.

```suggestion
    <description>The Sider's recommended ruleset extension is a set of useful rules except PSR12, and is intended to be used in combination with SiderRecPSR12.</description>
```

File: images/code_sniffer/sider_recommended_code_sniffer.xml (hunk 1 above)
Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: I think so too, and fixed it.

Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: I added it.

File: images/code_sniffer/sider_recommended_code_sniffer_ext.xml (hunk 2 above)
Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer_ext.xml:72]: I fixed it.

File: images/code_sniffer/sider_recommended_code_sniffer.xml (hunk 3 above)
Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: I removed it.

File: images/code_sniffer/sider_recommended_code_sniffer_ext.xml (hunk 4 above)
Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer_ext.xml]: I fixed it.

**Comment by sleeeto** [comment]
@ybiquitous Ready for merge.

File: images/code_sniffer/sider_recommended_code_sniffer.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/squizlabs/PHP_CodeSniffer/HEAD/phpcs.xsd" name="SiderPSR12">
```
Comment (ybiquitous) [inline images/code_sniffer/sider_recommended_code_sniffer.xml:2]: [suggestion] Does it make sense to make the code readable to add new lines as below?

```suggestion
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
         xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/squizlabs/PHP_CodeSniffer/HEAD/phpcs.xsd"
         name="SiderPSR12">
```

Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: OK.

**Comment by sleeeto** [comment]
@ybiquitous Ready.

File: images/code_sniffer/sider_recommended_code_sniffer.xml (hunk 5 above)
Comment (ybiquitous) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: I mean the attributes are easy to read if they are aligned vertically. What do you think?

```xml
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/squizlabs/PHP_CodeSniffer/HEAD/phpcs.xsd"
         name="SiderPSR12">

Comment (sleeeto) [reply images/code_sniffer/sider_recommended_code_sniffer.xml]: OK, fixed.

**Comment by sleeeto** [comment]
@ybiquitous Ready.

**Review by ybiquitous — APPROVED** [review]
👍

**Pull request merged**